### ISSUE TITLE
Investigate NTLM authentication on Linux for NegotiateStream

### ISSUE THREAD

**Issue by iamjasonp** [dotnet/wcf#943]
_Linux only_
We currently have turned on and verified support for Kerberos auth on Linux, we have yet to be able to support NTLM authentication. 

NegotiateStream relies on the underlying `gssapi` implementation for Kerberos and NTLM support. If that implementation supports NTLM fallback, then NegotiateStream can also do so. 

The following platforms are or will soon support NTLM fallback because their package managers install the `GSS-NTLMSSP` implementation of `gssapi`. 

* RedHat Enterprise Linux (RHEL) 
* CentOS

The following distros are known to currently not support NTLM as they package the MIT `libgssapi_krb5.so` library by default. 

* Ubuntu
* Debian 

If it's important to have NTLM fallback on these distros, it's possible to use GSS-NTLMSSP using the package manager or one can build it on their own. 

Note that this effectively means our test matrix for NegotiateStream/Kerberos/NTLM has an additional pivot: the implementation of `gssapi` - in this case, 

* GSS-NTLMSSP, or 
* MIT 

We currently do not have testing here; we will need to start thinking about testing this soon.

**Comment by zhenlan** [comment]
@iamjasonp looks like there two parts of this issue 1) add tests for the scenarios we did not test yet 2) make this as documentation (.md). Is this your plan?

**Comment by iamjasonp** [comment]
Yes, this sounds good. Also, we should include the table from #898 as well.

**Comment by kristofferjansson** [comment]
@iamjasonp - Hello, is the NTLM support for Linux something that you have investigated any more?

**Comment by Lxiamail** [comment]
Looks like the blocking issue has been fixed. We should check what's the current status of this work item.

**Comment by StephenBonikowsky** [comment]
As I currently understand it the remaining work here is to add scenario test coverage for NTLM Auth on Linux but based on associated Issues there is some difficulty automating this particular type of test. Need to do some additional investigation to fully understand the problems.

**Comment by StephenBonikowsky** [comment]
@mconnew 
We need to understand the exact state of this feature before we know what the next step is.
